src/CoreFoundation/CFReadStream.cs

```diff
@@ -151,14 +151,8 @@ public nint Read (byte[] buffer, int offset, int count)
 				throw new ArgumentException ();
 			if (offset + count > buffer.Length)
 				throw new ArgumentException ();
-			var gch = GCHandle.Alloc (buffer, GCHandleType.Pinned);
-			try {
-				var addr = gch.AddrOfPinnedObject ();
-				var ptr = new IntPtr (addr.ToInt64 () + offset);
-				return CFReadStreamRead (Handle, ptr, count);
-			} finally {
-				gch.Free ();
-			}
+			fixed (byte* ptr = &buffer [0])
```

Comment:
buffer.Length can't be 0 here due to argument validation (count has to be >= 1, and offset + count has to be <= buffer.Length).